### ISSUE TITLE
MongoDB v6.0.2 or mongoplayground bug?

### ISSUE THREAD

**Issue by rickhg12hs** [feliixx/mongoplayground#153]
While exploring ways to retrieve or `"$merge"`/`"$out"` unique documents from a collection, I ran across the following "oddity".

[mongoplayground.net example](https://mongoplayground.net/p/fd7qKuKRAAB)

mongoplayground `mgodatagen` configuration:
```json
[
  {
    "collection": "collection",
    "count": 100,
    "content": {
      "key1": {
        "type": "int",
        "minInt": 0,
        "maxInt": 1
      },
      "key2": {
        "type": "int",
        "minInt": 0,
        "maxInt": 1
      }
    }
  }
]
```
Query:
```js
db.collection.aggregate([
  {
    "$unset": "_id"
  },
  {
    "$group": {
      "_id": "$$ROOT"
    }
  },
  {
    "$replaceWith": "$_id"
  }
])
```
Result:  (should just be 4 docs, but it's not - there are repeats)
```json
[
  {
    "key1": 1,
    "key2": 1
  },
  {
    "key1": 1,
    "key2": 0
  },
  {
    "key1": 1,
    "key2": 0
  },
  {
    "key1": 1,
    "key2": 1
  },
  {
    "key1": 0,
    "key2": 0
  },
  {
    "key1": 0,
    "key2": 1
  },
  {
    "key1": 0,
    "key2": 0
  },
  {
    "key1": 0,
    "key2": 1
  }
]
```
Using the same collection and query on MongoDB Atlas, the expected output of just 4 documents was produced.

The MongoDB Atlas Cluster I used is at version 5.0.13 so maybe it's a MongoDB server regression?

**Comment by rickhg12hs** [comment]
I have another data point, I think.

TLDR; I think object field ordering isn't preserved when creating the MongoDB collection.

Rather than splash some more configs and queries here, I just supply the mongoplayground.net links.

Here's a [mongoplayground.net](https://mongoplayground.net/p/72KUBAzBOWT) where the results are not consistent.  Hitting <kbd>run</kbd> several times, sometimes the output is correct (essentially converting an array to a set) and sometimes the result has repeat elements in the array.

Here, the [mongoplayground.net](https://mongoplayground.net/p/596CWCJv8zL) has the same query, _except_ the objects in the array are rewritten.  The result is consistent and correct.

Does this mean that documents written to the collection have randomly/inconsistently ordered objects?

This would also explain the "oddity" described in my first comment above.

**Comment by feliixx** [comment]
> TLDR; I think object field ordering isn't preserved when creating the MongoDB collection.

Exactly. It's exactly what's happening here, for example in `mongosh`

```js
use test
db.test.insertOne({a:1,b:0})
db.test.insertOne({b:0,a:1})
db.test.aggregate([ { "$unset": "_id" }, { "$group": { "_id": "$$ROOT" } }, { "$replaceWith": "$_id" }])
```
Result: 
`[ { a: 1, b: 0 }, { b: 0, a: 1 } ]`

Sadly, it's the driver that's responsible of this. It's actually the same root cause as #150... So another good reason to try to fix this field reordering issue!

**Comment by johachi** [comment]
Could this also be related to https://github.com/feliixx/mongoplayground/issues/171 regarding getting inconsistent results?
